app/controllers/rounds_controller.rb

```diff
@@ -20,45 +20,96 @@ def view_pairings
-                      pairings: round.pairings.map do |pairing|
-                                  {
-                                    id: pairing.id,
-                                    table_number: pairing.table_number,
-                                    table_label: pairing.table_label,
-                                    policy: {
-                                      view_decks:
-                                    },
-                                    player1: pairing_player1(stage, pairing),
-                                    player2: pairing_player2(stage, pairing),
-                                    score_label: score_label(pairing),
-                                    intentional_draw: pairing.intentional_draw,
-                                    two_for_one: pairing.two_for_one
-                                  }
-                                end,
-                      pairings_reported: round.pairings.select { |p| p.score1 && p.score2 }.count
-                    }
-                  end
+      is_player_meeting: @tournament.round_ids.empty?,
+      stages: []
+    }
+
+    pairings_fields = %i[id table_number player1_id player2_id side intentional_draw
+                         two_for_one score1 score1_corp score1_runner score2 score2_corp score2_runner]
+
+    stages.map do |stage|
+      view_decks = stage.decks_visible_to(current_user) ? true : false
+      stage_out = {
+        name: stage.format.titleize,
+        format: stage.format,
+        rounds: []
+      }
+      stage.rounds.each do |r|
+        round = {
+          id: r.id,
+          number: r.number,
+          pairings: [],
+          pairings_reported: 0
         }
+
+        r.pairings.order(:table_number).pluck(pairings_fields).each do | # rubocop:disable Metrics/ParameterLists
+            id, table_number, player1_id, player2_id, side, intentional_draw,
+            two_for_one, score1, score1_corp, score1_runner, score2, score2_corp, score2_runner|
+          round[:pairings_reported] += score1.nil? && score2.nil? ? 0 : 1
+          player1 = players[player1_id]
+          player2 = players[player2_id]
+          round[:pairings] << {
+            id:,
+            table_number:,
+            table_label: stage.double_elim? ? "Game #{table_number}" : "Table #{table_number}",
+            policy: {
+              view_decks:
+            },
+            player1: {
+              name_with_pronouns: name_with_pronouns(player1),
+              side: player1_side(side),
+              side_label: player1_side_label(side),
+              corp_id: corp_id(player1),
+              runner_id: runner_id(player1)
+            },
+            player2: {
+              name_with_pronouns: name_with_pronouns(player2),
+              side: player2_side(side),
+              side_label: player2_side_label(side),
+              corp_id: corp_id(player2),
+              runner_id: runner_id(player2)
+            },
```

Comment:
This code to process the players looks like more duplication than I'm really comfortable with, with both the structure here and the methods below being duplicated. We might make one method to render a player if we choose the parameters carefully.